web_src/css/modules/button.css

```diff
@@ -67,12 +67,12 @@ It needs some tricks to tweak the left/right borders with active state */
 }
 
 .ui.buttons .button:first-child,
-.ui.buttons .button.gt-hidden:first-child + .button {
+.ui.buttons .button.tw-hidden:first-child + .button {
   border-left: 1px solid var(--color-light-border);
 }
 
 .ui.buttons .button:last-child,
-.ui.buttons .button:nth-last-child(2):has(+ .button.gt-hidden) {
+.ui.buttons .button:nth-last-child(2):has(+ .button.tw-hidden) {
```

Comment:
I really dislike such unclear code. No test for it, not clear what it is used for.

The new code is from yesterday's #30014 ........

I ever tried to make the "buttons" clear and easy to use (#26891, with UI demos), I don't want to use too many hacky patches for buttons again.

Reply:
`last-child` matches on hidden elements as well, so without this selector had hidden the border on the invisible button and the second-to-last item had a missing border. Now with this it works perfectly.

I agree the CSS on button groups is a hack and needs to be rewritten.

Reply:
For example, why you only added `nth-last-child(2)`, but not `nth-last-child(3)` even `nth-last-child(4)`? Many buttons could be "hidden".

For such case, maybe JS should be used to avoid such JS tricks.

Reply:
> but first we should migrate the `button` module from fomantic into first-party code.

Which page/code needs such trick? I guess only one or two places at most, it's not worth to make the "buttons" that complex.

Reply:
Blame view, it's the only place I remember having such a button group:

<img width="442" alt="image" src="https://github.com/go-gitea/gitea/assets/115237/1d99ccea-c767-40c7-aac9-18fe6b9730cd">